client/my-sites/current-site/notice.jsx

```diff
@@ -92,7 +92,7 @@ const SiteNotice = React.createClass( {
 
 export default connect( ( state, ownProps ) => {
 	return {
-		hasDomainCredit: hasDomainCredit( state, ownProps.site.ID )
+		hasDomainCredit: !! ownProps.site && hasDomainCredit( state, ownProps.site.ID )
```

Comment:
It's a common js usage. I think its a bit more clear for !! to denote that we're testing for truthyness. The Boolean conversion can sometimes return unexpected things, like so:
```
Boolean("false"); // still returns true 
```